webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/properties/MergeNodeProperties.scala

```diff
@@ -398,43 +406,49 @@ object MergeNodeProperties {
+      val grouped = propByTrees.groupByNec(_.prop.name)
+
+      // The right combinator that allows to use the semigroup structure of the error is the
+      val validatedProps = grouped.toList.map {
+        case (k, c) =>
+          val (h, more) = (c.head, c.tail)
+          if (more.forall(_.prop == h.prop)) { // if it's exactly the same property everywhere, it's ok
             Right(h)
           } else {
-            // faulty groups are the head of parent list of each parent list
-            val faulty = (h :: more).map { p =>
-              p.hierarchy match {
-                case Nil          => "" // should not happen
-                case g :: Nil     =>
-                  g.displayName
-                case g :: parents =>
-                  s"${g.displayName} (with inheritance from: ${parents.map(_.displayName).mkString("; ")}"
-              }
-            }
             Left(
-              Inconsistency(
-                s"Error when trying to find overrides for group property '${k}'. " +
-                s"Several groups which are not in an inheritance relation define it. You will need to define " +
-                s"a new group with all these groups as parent and choose the order on which you want to do " +
-                s"the override by hand. Faulty groups: ${faulty.mkString(", ")}"
-              )
+              NodePropertyError.PropertyInheritanceConflicts
+                .one(
+                  h.prop,
+                  NonEmptyChunk(h.hierarchy) ++ Chunk.from(more.map(_.hierarchy).iterator)
+                )
             )
           }
       }
-      grouped.accumulatePure(identity)
+
+      // This is guaranteed to be at least a Both, since we start with a right one and we just add potential errors
+      val initialValue = Ior.right[NodePropertyError.PropertyInheritanceConflicts, Chain[NodePropertyHierarchy]](Chain.empty)
+      validatedProps
+        .foldLeft(initialValue) {
+          case (acc, Left(c))  => acc.addLeft(c)
+          case (acc, Right(n)) => acc.addRight(Chain.one(n))
+        }
+        .map(_.toList)
     }
 
     for {
-      groups    <- targets.traverse(_.toGroupProp.map(g => (g.groupId.serialize, g))).map(_.toMap)
-      sorted    <- sortGroups(groups)
+      sorted    <- Ior.fromEither(sortGroups(props.map { case (k, v) => k.serialize -> v })).addRight {
+                     // Return an empty list of custom group properties instead of failing with the no success properties at all.
+                     // The consequence would be that it would seem like there are no groups at all, so nodes directly inherit global parameters.
+                     // (later we could even return a sort by property that for example adds custom properties for resolved sub-dag,
+                     // despite a global group DAG error...)
+                     List.empty[List[GroupProp]]
+                   }
```

Comment:
if it's the case, then the algo must be change, this is not what we want: 
- global properties are known and out of inheritance, they must alway be resolved. We can't loose information in the process,
- if the DAG resolution fails, we can't know what the value is. It's impossible: the inheritage is broken. Giving a value in that case means that we are just guessing, or worse, ignoring an attempt of changing the inheritage. It must be an error, else the model lies for what we are trying to understand.
- but: we still know the value defined on the group, which can be displayed on UI or error log and must be kept, 
- and the DAG resolution failure is property by property: we can have some properties in a broken DAG, and other in a correctly resolved one. We must not fail the resolution of all properties because one is in error. 

So I think the data structure may need a group specific error similar to nodes. 
And the sort should be done property by property (I'm not sure we can easily find how to group properties together, it looks like we could just do one resolution by intersection set of properties, but I'm not sure it's faster to compute lots of intersections, nor that it's mathematically sound).

Reply:
The impact on generation is [there](https://github.com/Normation/rudder/blob/c752aa9ff9fad1e2dc7396901ff9cd1886e86e4/webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala#L923) : 
* we consider that any state of error in the resolution of properties stored in the cache leads to the property being unresolved, and thus to global properties being ignored on each node with the error during policy generation
* we still use the stored error in the cache to get the message or the error and return it via API (frontend consumers have different usage that policy generation)

Reply:
:white_check_mark:  This is now failing at generation when it's not a success : https://github.com/Normation/rudder/pull/5910/commits/6223dafde5801eec7fc930716f24054115fa25e4#diff-bcf65e60a16e1339bef176122eda5d97d71066814f0c22c6e43acd20e354706aR1064-R1072